lib/rules/lint-no-action.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const Rule = require('./base');
+
+const ERROR_MESSAGE = 'Do not use `action` as %. Instead, use the `on` modifier and `fn` helper.';
+
+module.exports = class NoAction extends Rule {
+  visitor() {
+    const isLocal = this.isLocal.bind(this);
+    const log = this.log.bind(this);
+    const sourceForNode = this.sourceForNode.bind(this);
+    let closestTag = null;
+
+    function detectAction(node, usageContext) {
+      if (isLocal(node.path)) {
+        return;
+      }
+      let maybeAction = node.path.original;
+      if (maybeAction !== 'action') {
+        return;
+      }
+      if (node.path.data === true || node.path.this === true) {
```

Comment:
Doesn't the `node.path.original` guard just above handle both of these cases? For `{{@action}}` (the `node.path.data` check) and `{{this.action}}` (the `node.path.this` check) `node.path.original` should not be `"action"`...